lib/OperatingSystems/IOperatingSystem.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2020 Daniel Kesselberg <mail@danielkesselberg.de>
+ *
+ * @author Daniel Kesselberg <mail@danielkesselberg.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\ServerInfo\OperatingSystems;
+
+use OCA\ServerInfo\Resources\Memory;
+
+interface IOperatingSystem {
+	/**
+	 * Get memory returns a Memory object. All values are in bytes.
+	 *
+	 * @return Memory
+	 */
+	public function getMemory(): Memory;
+
+	/**
+	 * Get name of the processor
+	 *
+	 * @return string
+	 */
+	public function getCPUName(): string;
```

Comment:
```suggestion
	public function getCpuName(): string;
```

debatable, but I think that reads better

Reply:
Fine by me.